addon/-private/system/diff-array.js

```diff
@@ -57,3 +58,95 @@ export default function diffArray(oldArray, newArray) {
     removedCount
   };
 }
+
+export function setForArray(array) {
+  let set = new OrderedSet();
+
+  if (array) {
+    for (let i=0, l=array.length; i<l; i++) {
+      set.add(array[i]);
+    }
+  }
+
+  return set;
+}
+
+export function computeChanges(oldArray, newArray, removalSet) {
```

Comment:
let's document the semantics of this function

Reply:
speaking of which we should probably not have it in `diff-array` as it's semantics are more specific than array diffing

if we don't want to keep it in `relationships/state/has-many` maybe move it to something like `relationships/state/utils` and perhaps rename it to `computeHasManyChanges(previousCanonicalState, newCanonicalState, locallyRemoved)`